scaladoc/src/dotty/tools/scaladoc/tasty/comments/Preparser.scala

```diff
@@ -35,11 +35,13 @@ object Preparser {
       inCodeBlock: Boolean
     ): PreparsedComment = remaining match {
       case CodeBlockStartRegex(before, marker, after) :: ls if !inCodeBlock =>
-        if (!before.trim.isEmpty && !after.trim.isEmpty)
+        if (before.trim.nonEmpty && after.trim.nonEmpty && marker == "```")
+          go(docBody, tags, lastTagKey, before :: (marker + after) :: ls, inCodeBlock = false)
+        else if (before.trim.nonEmpty && after.trim.nonEmpty)
           go(docBody, tags, lastTagKey, before :: marker :: after :: ls, inCodeBlock = false)
-        else if (!before.trim.isEmpty)
+        else if (before.trim.nonEmpty)
```

Comment:
Those were probably intentional. The JIT has a harder time optimizing calls to `nonEmpty` as it contains a metamorphic call to `isEmpty`.

Reply:
Ooo, in that case- changed back to `.isEmpty`